e2etest/pkg/container/routes.go

```diff
@@ -93,3 +98,140 @@ func SetupVRFForNetwork(containerName, vrfNetwork, vrfName, vrfRoutingTable stri
 
 	return nil
 }
+
+// WireContainers creates a point-to-point link between two containers.
+// The second argument will be the name of the interface which will be
+// common to both cotnainers e.g. net0.
+func WireContainers(containerA, containerB, dev string) error {
+	netNSCntA, err := exec.Command(executor.ContainerRuntime,
+		"inspect", "-f", "{{ .NetworkSettings.SandboxKey }}", containerA).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%s - %w", netNSCntA, err)
+	}
+	netNSCntA = bytes.TrimSpace(netNSCntA)
+
+	netNSCntB, err := exec.Command(executor.ContainerRuntime,
+		"inspect", "-f", "{{ .NetworkSettings.SandboxKey }}", containerB).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%s - %w", netNSCntB, err)
+	}
+	netNSCntB = bytes.TrimSpace(netNSCntB)
+
+	c := fmt.Sprintf("ip link add %s netns %s type veth peer name %s", dev, netNSCntA, dev)
+	if out, err := executor.Host.Exec("sudo", strings.Split(c, " ")...); err != nil {
+		return fmt.Errorf("%s - %w", out, err)
+	}
+
+	c = fmt.Sprintf("ip link set dev %s netns %s", dev, netNSCntB)
+	if out, err := executor.Host.Exec("sudo", strings.Split(c, " ")...); err != nil {
+		return fmt.Errorf("%s - %w", out, err)
+	}
+
+	node := executor.ForContainer(containerB)
+	if out, err := node.Exec("ip", "link", "set", "dev", dev, "up"); err != nil {
+		return fmt.Errorf("%s - %w", out, err)
+	}
+
+	peer := executor.ForContainer(containerA)
+	if out, err := peer.Exec("ip", "link", "set", "dev", dev, "up"); err != nil {
+		return fmt.Errorf("%s - %w", out, err)
+	}
+
+	return nil
+}
+
+// GetIPv6Link return the LinkLocalAddress (LLA) of a specific interface.
+func GetIPv6Link(containerName, dev string) (string, error) {
+	exc := executor.ForContainer(containerName)
+	return netdev.LinkLocalAddressForDevice(exc, dev)
+}
+
+// BGPRoutes executes `ip route` command to fetch BGP routes
+// and return a map of map leveraging netip.{Prefix,Addr} which
+// allow direct comparison.
```

Comment:
can you make it clearer that what is returned is a map of routes -> their set of nexthops, and the role of "dev" here? also, using bool instead of struct{} is a bit cleaner

Reply:
right, it's not something critical, just what we did in other places in the code iirc (also, did you try using `k8s.io/apimachinery/pkg/util/sets` for this?).
also, the:
```
// The return json from the kernel varies depend if ipv4 with single entry, or
// ipv4 with multiple entries or ipv6. The use of maps, and using keys the types
// netip.{Prefix,Addr} allows direct comparison.
```
part is unnecessary

Reply:
Removed the part. I have not really tried to use the set from the k8s.io. I think the test function is okay as is using the standard lib but now strong opinion. Would you like that I refactor to use the `k8s.io/apimachinery/pkg/util/set`?

Reply:
not a big deal, we can leave it as is